### ISSUE TITLE
Adpative cell length when formatting GFM tables

### ISSUE THREAD

**Issue by yzhang-gh** [yzhang-gh/vscode-markdown#1233]
### Proposal

when the total table width is too large (e.g., >120), try to save space

#### Case 1

```markdown
| a   | b         | c          |
| --- | --------- | ---------- |
| a   | b         | ccc        |
| a   | bbbbbbbbbbbbbbbb | c   |
| a   | b         | cccccc     |
| a   | b         | cccccccccc |
```

instead of

```markdown
| a   | b                | c          |
| --- | ---------------- | ---------- |
| a   | b                | ccc        |
| a   | bbbbbbbbbbbbbbbb | c          |
| a   | b                | cccccc     |
| a   | b                | cccccccccc |
```

#### Case 2

```markdown
| a   | b    | c   |
| --- | ---- | --- |
| a   | b    | cc  |
| a   | bbbb | c   |
| a   | bb   | ccc |
| a   | bbb  | ccccccccccccc |
```

instead of

```markdown
| a   | b    | c             |
| --- | ---- | ------------- |
| a   | b    | cc            |
| a   | bbbb | c             |
| a   | bb   | ccc           |
| a   | bbb  | ccccccccccccc |
```

**Comment by physcrowley** [comment]
Or even as a switch to skip formatting that extra wide table.

I have this issue a lot when inserting image urls or descriptions into a table cell and the formatter completely ruins the document layout unless I turn Word Wrap off in my editor.

unformatted:
![image](https://github.com/user-attachments/assets/5d8ced6d-1193-437e-b186-203e04d2bf70)

formatted disaster:
![image](https://github.com/user-attachments/assets/4b05ba75-5ba5-4749-8620-ddd4dea42fd6)

word wrap off after formatting:
![image](https://github.com/user-attachments/assets/899d3ffe-6acb-49c4-8fb0-7e21bd6d1985)

**Comment by stefanb** [comment]
Allow extra long (above some default, adjustable limit) lines to NOT be aligned with others, while other are formatted aligned, eg

## maxTableLineLenght=100 (default)
```markdown
| a a | b                                                      | c          |
| --- | ------------------------------------------------------ | ---------- |
| a   | b                                                      | ccc        |
| a   | bbb bbbbbbbbbbbbbbbbbbbbbbb bbbbbbbbbbbbbbbbb bbb bbbb | c          |
| a   | b                                                      | cccccc     |
| a   | b                                                      | cccccccccc |
```
Note it takes just what it needs, capped at 77 characters in this case. The result is the same for any maxTableLineLenght>=77

## maxTableLineLenght=70)
```markdown
| a a | b                                               | c          |
| --- | ----------------------------------------------- | ---------- |
| a   | b                                               | ccc        |
| a   | bbb bbbbbbbbbbbbbbbbbbbbbbb bbbbbbbbbbbbbbbbb bbb bbbb | c   |
| a   | b                                               | cccccc     |
| a   | b                                               | cccccccccc |
```
If needed adjust individual column widths to match others as much as possible, but to stay within set limit.

## maxTableLineLenght=40
```markdown
| a a | b                | c          |
| --- | ---------------- | ---------- |
| a   | b                | ccc        |
| a | bbb bbbbbbbbbbbbbbbbbbbbbbb bbbbbbbbbbbbbbbbb bbb bbbb | c |
| a   | b                | cccccc     |
| a   | b                | cccccccccc |
```

## maxTableLineLenght=25
```markdown
| a a | b | c          |
| --- | - | ---------- |
| a   | b | ccc        |
| a | bbb bbbbbbbbbbbbbbbbbbbbbbb bbbbbbbbbbbbbbbbb bbb bbbb | c |
| a   | b | cccccc     |
| a   | b | cccccccccc |
```

## maxTableLineLenght=20
```markdown
| a a | b | c     |
| --- | - | ----- |
| a   | b | ccc   |
| a | bbb bbbbbbbbbbbbbbbbbbbbbbb bbbbbbbbbbbbbbbbb bbb bbbb | c |
| a  | b | cccccc |
| a | b | cccccccccc |
```
When limit is set extremely low all excessively long lines are left unaligned, with no extra padding

Basically, try to align the table columns within set line length limit (with some reasonable default). If the limit cannot be met just compact the line.
